iocore/net/UnixNetVConnection.cc

```diff
@@ -336,7 +330,12 @@ read_from_net(NetHandler *nh, UnixNetVConnection *vc, EThread *thread)
         return;
       }
     }
+  } else if (vc->has_error()) {
```

Comment:
`read_from_net()` is a pretty long method and I'm confused. We come here only if the `toread` is `0`?

Reply:
I concern this raise `VC_EVENT_ERROR` in unexpected timing like
- the reading buffer is full (`toread == 0`)
- the peer sent all data and reset the connection

Reply:
Hmm.  The intent was to issue the read error if the reading had not otherwise hit an error, but that is not what we're doing here.  You are correct.

Probably best to remove the error sending on the read side, but leave it on the write side (from the original) PR.  I'll updated.